fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -1073,15 +1082,28 @@ struct MetaclusterManagementWorkload : TestWorkload {
 				return Void();
 			} else if (e.code() == error_code_cluster_no_capacity) {
 				ASSERT(!assignClusterAutomatically);
+				// When trying to create a new tenant, there are several cases in which "cluster_no_capacity" can be
+				// thrown. The only one case in which ignoreCapacityLimit is true is when the number of tenants exceed
+				// a threshold configured via knob. But this is not possible due to the way we pick tenant names in this
+				// test, thus we can have the following assertion.
+				ASSERT(!ignoreCapacityLimit);
 				return Void();
 			} else if (e.code() == error_code_cluster_not_found) {
 				ASSERT(!assignClusterAutomatically);
 				return Void();
 			} else if (e.code() == error_code_invalid_tenant_configuration) {
-				ASSERT(tenantGroup.present());
-				auto itr = self->tenantGroups.find(tenantGroup.get());
-				ASSERT(itr != self->tenantGroups.end());
-				ASSERT(itr->second->cluster != tenantMapEntry.assignedCluster);
+				if (tenantGroup.present()) {
+					auto itr = self->tenantGroups.find(tenantGroup.get());
+					if (itr != self->tenantGroups.end()) {
+						ASSERT(itr->second->cluster != tenantMapEntry.assignedCluster);
+					} else {
+						ASSERT(assignClusterAutomatically);
+						ASSERT(ignoreCapacityLimit);
+					}
+				} else {
+					ASSERT(assignClusterAutomatically);
+					ASSERT(ignoreCapacityLimit);
+				}
```

Comment:
I think this will need to be written a little differently in light of some of the other proposed changes. You could have something like:

```
bool tenantGroupClusterMismatch = ...
bool invalidIgnoreCapacityLimit = ...
ASSERT(tenantGroupClusterMismatch || invalidIgnoreCapacityLimit);
```

Reply:
Done